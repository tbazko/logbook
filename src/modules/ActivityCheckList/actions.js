import * as types from 'types'

export const toggleItemCheckbox = (id, createdAt) =>
  ({ type: types.ActivityCheckList.TOGGLE_ITEM_CHECKBOX, id, createdAt })

export const removeListItem = (id, activityType) =>
  ({ type: types.ActivityCheckList.REMOVE_LIST_ITEM, id, activityType })
export const removeAll = () => ({ type: types.ActivityCheckList.REMOVE_ALL })


export const checkAll = timestamp => ({ type: types.ActivityCheckList.CHECK_ALL, timestamp })
export const uncheckAll = timestamp => ({ type: types.ActivityCheckList.UNCHECK_ALL, timestamp })
export const setDefaultCheckboxValue = (timestamp, activityTypes) =>
  ({ type: types.ActivityCheckList.SET_DEFAULT_VALUES, timestamp, activityTypes })

export const setActiveCheckList = timestamp => ({
  type: types.ActivityCheckList.SET_ACTIVE_CHECKLIST,
  timestamp,
})

