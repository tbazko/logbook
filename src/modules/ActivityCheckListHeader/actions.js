import * as types from 'types'
import moment from 'moment'

export const showPrevDate = timestamp => ({
  type: types.ActivityCheckListHeader.SHOW_PREV_DATE,
  timestamp: moment(timestamp).add(-1, 'days').startOf('day').format(),
})

export const showNextDate = (timestamp, activityTypes) => ({
  type: types.ActivityCheckListHeader.SHOW_NEXT_DATE,
  timestamp: moment(timestamp).add(1, 'days').startOf('day').format(),
  activityTypes,
})
