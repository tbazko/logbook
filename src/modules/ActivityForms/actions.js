import uuid from 'react-native-uuid'
import moment from 'moment'
import * as types from 'types'

export const addItem = title => ({
  type: types.ActivityForm.ADD_ACTIVITY_TYPE,
  id: uuid.v1(),
  title,
  createdAt: moment().startOf('day').format(),
})

export const editItem = (id, payload) => ({
  type: types.ActivityForm.EDIT_ACTIVITY_TYPE,
  id,
  payload,
})

export const setViewMode = viewMode => ({
  type: types.ActivityCheckListHeader.SET_VIEW_MODE,
  viewMode,
})

export const addItemError = error => ({ type: types.ActivityForm.ADD_ACTIVITY_ERROR, error })
export const removeItemError = () => ({ type: types.ActivityForm.REMOVE_ACTIVITY_ERROR })
