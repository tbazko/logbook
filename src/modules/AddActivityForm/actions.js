import uuid from 'react-native-uuid'
import moment from 'moment'
import * as types from 'types'

export const addItem = title => ({
  type: types.AddActivityForm.ADD_ACTIVITY_TYPE,
  id: uuid.v1(),
  title,
  timestamp: moment().startOf('day').unix(),
})
export const addItemError = error => ({ type: types.AddActivityForm.ADD_ACTIVITY_ERROR, error })
