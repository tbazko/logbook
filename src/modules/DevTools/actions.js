import moment from 'moment'
import * as types from 'types'

export const resetStore = () => ({
  type: types.DevTools.REMOVE_ALL_DATA,
})

const todayTS = moment().startOf('day').format()
const activityTypes = {
  'Yoga-Id': {
    title: 'Yoga',
  },
  'Sport-Id': {
    title: 'Sport',
  },
  'Vega-Id': {
    title: 'Vega',
  },
  'Meat-Id': {
    title: 'Meat',
  },
  'Alco-Id': {
    title: 'Alco',
  },
}

export const insertData = () => ({
  type: types.DevTools.INSERT_DATA,
  payload: {
    activityTypes,
    activityLogs: getActivityLogs(),
    activeLogTimestamp: todayTS,
  },
})

export function getActivityLogs() {
  const logs = {}

  for (let i = 0; i < 60; i++) {
    logs[moment(todayTS).subtract(i, 'day').format()] = {
      'Sport-Id': {
        completed: parseInt(Math.random() * 10, 10) > 5,
      },
      'Vega-Id': {
        completed: parseInt(Math.random() * 10, 10) > 5,
      },
      'Meat-Id': {
        completed: parseInt(Math.random() * 10, 10) > 5,
      },
      'Alco-Id': {
        completed: parseInt(Math.random() * 10, 10) > 5,
      },
      'Yoga-Id': {
        completed: parseInt(Math.random() * 10, 10) > 5,
      },
    }
  }

  return logs
}
