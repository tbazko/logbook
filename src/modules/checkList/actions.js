import * as types from './constants'

export const toggleListItem = id => ({ type: types.TOGGLE_LIST_ITEM, id });
export const toggleAll = () => ({ type: types.TOGGLE_ALL_LIST_ITEMS });
export const addListItem = title => ({ type: types.ADD_LIST_ITEM, title });
