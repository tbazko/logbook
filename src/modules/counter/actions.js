import * as types from './constants';


export const incrementCounter = () => ({ type: types.INCREMENT_COUNTER });
export const decrementCounter = () => ({ type: types.DECREMENT_COUNTER });
export const clearCounter = () => ({ type: types.RESET_COUNTER });
