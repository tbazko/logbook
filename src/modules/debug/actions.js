import * as types from 'types'
import data from 'storeMocks/activities.mock'

export const insertData = () => {
  console.log('tamara dispatching', data)
  return {
    type: types.Debug.INSERT_DATA,
    payload: data,
  }
}
