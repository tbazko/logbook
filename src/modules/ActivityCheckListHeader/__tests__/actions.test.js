import moment from 'moment'
import mockStore from 'redux-mock-store'
import * as action from '../actions'

describe('ActivityForm actions', () => {
  let store

  beforeEach(() => {
    store = mockStore()
  })

  it('should dispatch SHOW_PREV_DATE with timestamp and activityTypes object', async () => {
    await store.dispatch(action.showPrevDate(moment.unix(1518303600).format()))
    expect(store.getActions()).toMatchSnapshot()
  })

  it('should dispatch SHOW_NEXT_DATE with timestamp and activityTypes object', async () => {
    await store.dispatch(action.showNextDate(moment.unix(1518303600).format(), { 'Sport-Id': { title: 'Sport' } }))
    expect(store.getActions()).toMatchSnapshot()
  })
})
