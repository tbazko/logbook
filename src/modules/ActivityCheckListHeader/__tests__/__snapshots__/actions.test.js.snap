// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityForm actions should dispatch SHOW_NEXT_DATE with timestamp and activityTypes object 1`] = `
Array [
  Object {
    "activityTypes": Object {
      "Sport-Id": Object {
        "title": "Sport",
      },
    },
    "timestamp": "2018-02-12T00:00:00+01:00",
    "type": "logbook.ActivityCheckListHeader.SHOW_NEXT_DATE",
  },
]
`;

exports[`ActivityForm actions should dispatch SHOW_PREV_DATE with timestamp and activityTypes object 1`] = `
Array [
  Object {
    "timestamp": "2018-02-10T00:00:00+01:00",
    "type": "logbook.ActivityCheckListHeader.SHOW_PREV_DATE",
  },
]
`;
